crates/wasmtime/src/module.rs

```diff
@@ -300,11 +301,21 @@ impl Module {
     }
 
     unsafe fn compile(engine: &Engine, binary: &[u8]) -> Result<Self> {
-        let compiled = CompiledModule::new(
-            engine.compiler(),
-            binary,
+        #[derive(Hash)]
+        struct Env<'a>(&'a Compiler, &'a [u8]);
```

Comment:
Is the `Env` wrapper here needed? I think a tuple of hashtable types shoudl be hashable here